### DISCUSSION TITLE
woodpecker server listens on 80/443 ports even though not configured

### DISCUSSION THREAD

**Discussion by michael-brade** [woodpecker-ci/woodpecker#2697] [Ask a Question]
Hi, I have woodpecker-sever 1.0.3 running in Docker with docker-compose. I have used these variables:
```
  - WOODPECKER_HOST=https://woodpecker.$DOMAIN       # woodpecker needs to know its own public address
  - WOODPECKER_SERVER_ADDR=:8000

  - WOODPECKER_GRPC_ADDR=:9000
  - WOODPECKER_GRPC_SECRET_FILE=/run/secrets/woodpecker-grpc-secret

  - WOODPECKER_LETS_ENCRYPT=false
```
And yet, it eats ports 80 and 443. `docker ps` outputs:
```
055860d992e0   woodpeckerci/woodpecker-server:v1.0.3       "/bin/woodpecker-ser…"   11 seconds ago   Up 8 seconds    80/tcp, 443/tcp, 8000/tcp, 9000/tcp          dev-woodpecker
```
Why is that? Is it a bug? Or can I do something about it? It prevents starting another that actually listens on port 80 with the error that "port is already in use".

**Comment by qwerty287** [comment]
Do you use any other functions like healthcheck or TLS?

**Comment by michael-brade** [comment]
I didn't enable any of those, so it should be no. The healthcheck seems to be only for the agent anyway, and I tried to explicitly disable TLS...

**Reply by qwerty287** [reply]
So I'm not an expert in docker, but probably it comes from this line:
https://github.com/woodpecker-ci/woodpecker/blob/c75068920cbd93f93931b0c1d6fbb3ed889a8746/docker/Dockerfile.server.multiarch#L10

Is there a different way that allows to expose docker ports at runtime and not at compile time (i.e. in docker-compose files)?

**Comment by michael-brade** [comment]
I have done some research exactly on this already. EXPOSE in Dockerfile is not really needed and of no practical use. Using -p or expose in docker-compose works. But I was wondering... if EXPOSE does nothing, the port should not actually be in use, so isn't woodpecker binding to it? Is it possible to find out who is binding to what port? ss and netstat don't seem to show any docker containers.

**Reply by michael-brade** [reply]
Oh, but I could probably just try to remove the EXPOSE line, rebuild the container and try again...

**Reply by qwerty287** [reply]
Did you try this? Did it work?

**Reply by michael-brade** [reply]
not yet, didn't have time yet. I think on Saturday I will be able to test this. (sorry, the days are full with stuff to work, this is "just" my own project)

**Reply by qwerty287** [reply]
Totally fine, we all maintain woodpecker in our free time :)

**Reply by michael-brade** [reply]
cool :) But well... can you believe it - it took me a while to figure out how to get it to build and when everything was done, I wanted to first check if the bug is still there. And it turns out I cannot reproduce it anymore :-( Busted...

**Reply by qwerty287** [reply]
So we can close this? If it's fixed it should be fine :)
